index.bs

```diff
@@ -322,7 +339,97 @@ spec:csp3; type:grammar; text:base64-value
   supported by this specification.
 
 
-  ## Response verification algorithms ## {#verification-algorithms}
+  ## Request verification algorithms ## {#request-verification-algorithms}
+
+  ### Opting-in  ### {#opt-in-require-sri-for}
+
+  Authors may opt a {{Document}} to require SRI metadata be present for 
+  some resource types via a <dfn export>require-sri-for</dfn> <a>Content
+  Security Policy</a> directive defined by the following ABNF grammar:
+
+  <pre dfn-type="grammar" link-type="grammar">
+      directive-name  = "require-sri-for"
+      directive-value = <a grammar>token</a> *( <a>OWS</a> <a>token</a> )
+  </pre>
+
+  The following list contains the set of <dfn noexport>known tokens</dfn>:
+
+    * `script` requires SRI for scripts
+
+
+  ### Parsing `require-sri-for` ### {#parse-require-sri-for}
+
+  Given a string (|token list|), this algorithm returns a list of resource
+  types which will require integrity checks:
+
+  1.  Let |protected resource types| be the empty <a for=/>set</a>.
+
+  2.  For each |token| in the result of <a lt="split a string on ascii whitespace">
+      splitting |token list| on spaces</a>, if token matches the grammar
+      for <a>require-sri-for</a> and is a <a>ASCII case-insensitive match</a>
+      for any of the <a>known token</a>s, add |token| to |protected resource types|.
+      Otherwise, ignore the token.
+
+  3.  Return the set of |protected resource types|.
+
```

Comment:
How likely is it that we expand the list of known tokens? Otherwise a simple string-equals match would suffice.

Reply:
Previously the list also included "style". As I don't have a clear use case for it, I didn't add it here.

We could drop this to be a simple equality check now, and expand to a list if/when the time comes.

Reply:
What should the behavior be in user agents which don't understand the new type?

This suggestion would make them not enforce anything at all, which seems suboptimal?

Reply:
I *think* this is the way CSP directives are currently defined, and that enabled us to expand them over time.

Not enforcing anything on unknown types would enable us to expand the list overtime if needed. What's the alternative you see?

Reply:
Thinking through @annevk's other comment, workers [may be](https://github.com/whatwg/html/pull/10858) a good candidate for near-future expansion of the list.